src/NuGet.Core/NuGet.Protocol/NullSourceCacheContext.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace NuGet.Protocol.Core.Types
+{
+    public class NullSourceCacheContext : SourceCacheContext
+    {
+        private static SourceCacheContext _instance;
+
+        public static SourceCacheContext Instance
+        {
+            get
+            {
+                if (_instance == null)
+                {
+                    _instance = new NullSourceCacheContext();
+                    _instance.DirectDownload = true;
+                }
+
+                return _instance;
+            }
+        }
+
+        public override string GeneratedTempFolder
+        {
+            get
+            {
+                return string.Empty;
```

Comment:
Is it possible this will get used? Should this return the env temp folder instead?

Reply:
It might be safer to throw if this is used, I'm not sure what would happen with empty, it might download to the working directory

obviously this instance shouldn't be used that way, but if it did happen

Reply:
after talking about this offline I think this should stay as is